main.go

```diff
@@ -177,23 +184,17 @@ func mapCRDAvailability(crdNames ...string) map[string]bool {
 
 	CRDExist := map[string]bool{}
 
-	for _, crdName := range crdNames {
-		crd := apiextensionsv1.CustomResourceDefinition{}
-		err = k8sClient.Get(context.Background(), types.NamespacedName{Name: crdName}, &crd)
+	crds := apiextensionsv1.CustomResourceDefinitionList{}
+	err = k8sClient.List(context.Background(), &crds)
 
-		if err != nil && !errors.IsNotFound(err) {
-			setupLog.Error(err, fmt.Sprintf("Error Getting CRD for %s", crdName))
-			os.Exit(1)
-		}
+	if err != nil && !errors.IsNotFound(err) {
+		setupLog.Error(err, fmt.Sprintf("Error Getting CRDs installed on the cluster"))
```

Comment:
The message here is fixed. no need for SprintF 
```suggestion
		setupLog.Error(err, "Error Getting CRDs installed on the cluster")
```